packages/mcdoc/src/runtime/checker/index.ts

```diff
@@ -445,15 +451,16 @@ export function typeDefinition<T>(
 	}
 
 	function attachTypeInfo(node: CheckerTreeRuntimeNode<T>) {
-		if (node.validDefinitions.length === 1) {
-			const { typeDef, desc } = node.validDefinitions[0]
-			options.attachTypeInfo(node.node.originalNode, typeDef, desc)
-			handleAttributes(typeDef.attributes, options.context, (handler, config) => {
+		function handleStringAttachers(
+			runtimeValue: RuntimeNode<T>,
+			typeDef: SimplifiedMcdocTypeNoUnion,
+		) {
```

Comment:
This function, and now that I think about it, maybe also `attachTypeInfo` could be moved out to the top level of the module. They only need an additional `options` parameter, they don't capture anything else right now.